code/datums/tutorial/marine/hospital_corpsman_sandbox.dm

```diff
@@ -0,0 +1,690 @@
+	// Will only roll beyond wave 3, and 3 waves after the previous resupply phase.
+	if(prob(20) && (survival_wave >= (last_resupply_round + 3)))
+		begin_supply_phase()
+		last_resupply_round = survival_wave
+		return
+	survival_wave++
+	// 1 in 10 chance per round to trigger a mass-cas (NIGHTMARE) phase
+	// Will only roll beyond wave 3, and 3 waves after the previous mass-cas phase.
+	if(prob(20) && (survival_wave >= (last_resupply_round + 3)))
+		stage = TUTORIAL_HM_PHASE_NIGHTMARE
+		// increases difficulty by 2 levels, but not beyond the max.
+		for(var/i in 1 to 2)
+			var/current_difficulty = survival_difficulty
+			if(current_difficulty != TUTORIAL_HM_INJURY_SEVERITY_MAXIMUM)
+				survival_difficulty = next_in_list(current_difficulty, difficulties)
+		// heightened patient NPC spawn rates, from 4-6
+		min_survival_agents = 4
+		max_survival_agents = 6
+		playsound(tutorial_mob.loc, 'sound/effects/siren.ogg', 50)
+		message_to_player("Warning! Mass-Casualty event detected!")
+	// 50% chance per wave of increasing difficulty by one step
+	// two round grace period from start
+	else if(prob(TUTORIAL_HM_DIFFICULTY_INCREASE * 100) && !(survival_wave <= 2))
+		var/current_difficulty = survival_difficulty
+		if(current_difficulty != TUTORIAL_HM_INJURY_SEVERITY_MAXIMUM)
+			survival_difficulty = next_in_list(current_difficulty, difficulties)
+			difficulty_upgrade_warning = " Difficulty has increased, watch out!!"
+
+	CMO_npc.say("Now entering round [survival_wave]![difficulty_upgrade_warning]")
+
+	addtimer(CALLBACK(src, PROC_REF(spawn_agents)), 2 SECONDS)
+
+/datum/tutorial/marine/hospital_corpsman_sandbox/proc/end_supply_phase()
+
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/structure/machinery/door/airlock/multi_tile/almayer/medidoor, prep_door)
+	var/turf/boundry = get_turf(loc_from_corner(4, 1))
+	if(tutorial_mob.x <= boundry.x)
+		message_to_player("Please exit the preperations room before progressing into the next round!")
+		return
+	prep_door.close(TRUE)
+	prep_door.lock(TRUE)
+	stage = TUTORIAL_HM_PHASE_MAIN
+	remove_action(tutorial_mob, /datum/action/hm_tutorial/sandbox/ready_up)
+	handle_round_progression()
+
+/datum/tutorial/marine/hospital_corpsman_sandbox/proc/begin_supply_phase()
+
+	TUTORIAL_ATOM_FROM_TRACKING(/obj/structure/machinery/door/airlock/multi_tile/almayer/medidoor, prep_door)
+	prep_door.unlock(TRUE)
+	prep_door.open()
+	stage = TUTORIAL_HM_PHASE_MAIN // just in case it wasnt already
+
+	message_to_player("Phew! We have entered a resupply phase of the tutorial!")
+	message_to_player("Use this rare opportunity to refill, restock, and resupply yourself for future rounds.")
+	message_to_player("Remember, on the field, immediate resupply will not always be possible! You won't know for certain when your next chance will arrive, so stock up while you can!")
+	message_to_player("When you are ready, leave the supply room, then click the 'Ready Up' action on the top left of your screen to begin your next round.")
```

Comment:
probably don't need to give this entire spiel after the 1st supply round they're in. Can just say "new resupply phase" or something simple after the first time